### PR TITLE
Introducing configurations for self registration with multiple channels

### PR DESCRIPTION
**Pull request by somindatommy**
### Proposed changes in this pull request
- Introducing a claim to track the users preferred channel of communication.
- Introducing configurations to support user self-registration via multiple channels.

Related to PR: https://github.com/wso2-extensions/identity-governance/pull/324

### CODE REVIEW

File: features/identity-event/org.wso2.carbon.identity.event.server.feature/resources/org.wso2.carbon.identity.event.server.feature.default.json [hunk 1]
```diff
@@ -119,5 +119,11 @@
   "identity_mgt.events.schemes.OIDCLogoutEventHandler.subscriptions": [
     "SESSION_TERMINATE"
   ],
+  "identity_mgt.events.schemes.'default.notification.sender'.module_index": "20",
```
Comment (pulasthi7) [inline features/identity-event/org.wso2.carbon.identity.event.server.feature/resources/org.wso2.carbon.identity.event.server.feature.default.json]: Need to update the j2 file to start the custom module indexes from 21 now

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 2]
```diff
@@ -245,6 +245,9 @@
   "identity_mgt.username_recovery.email.enable_username_recovery": false,
   "identity_mgt.username_recovery.email.enable_recaptcha": false,
 
+  "identity_mgt.notification.default_notification_channel" : "EMAIL",
+  "identity_mgt.notification.enable_resolve_notification_channel" : false,
```
Comment (pulasthi7) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]:
```suggestion
  "identity_mgt.notification.resolve_notification_channel" : false,
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 3]
```diff
@@ -874,6 +881,10 @@
 
     <SelfRegistration>
         <Enable>{{identity_mgt.user_self_registration.allow_self_registration}}</Enable>
+        <RegisterWithVerifiedChannels>{{identity_mgt.user_self_registration.register_with_verified_channels}}</RegisterWithVerifiedChannels>
```
Comment (pulasthi7) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: We may need to come up with a better name for this configuration

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 4]
```diff
@@ -874,6 +881,10 @@
...
+            <API>
```
Comment (pulasthi7) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Indentation

File: features/identity-event/org.wso2.carbon.identity.event.server.feature/resources/org.wso2.carbon.identity.event.server.feature.default.json (hunk 1 above)
Comment (somindatommy) [reply features/identity-event/org.wso2.carbon.identity.event.server.feature/resources/org.wso2.carbon.identity.event.server.feature.default.json]: This value is not hardcoded in the j2. The value is updated dynamically.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json (hunk 2 above)
Comment (somindatommy) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]: Fixed: d47fd742281a832a7570cade4cd44c63c87483b0

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 3 above)
Comment (somindatommy) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Fixed : d47fd742281a832a7570cade4cd44c63c87483b0

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 4 above)
Comment (somindatommy) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Fixed: d47fd742281a832a7570cade4cd44c63c87483b0

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 5]
```diff
@@ -253,6 +256,8 @@
   "identity_mgt.user_onboarding.password_generator": "org.wso2.carbon.user.mgt.common.DefaultPasswordGenerator",
 
   "identity_mgt.user_self_registration.allow_self_registration": false,
+  "identity_mgt.user_self_registration.skip_account_lock_on_verified_preferred_channel" : false,
```
Comment (pulasthi7) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]: avoid double negation

Comment (somindatommy) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]: Fixed: 0350af1c93f4c8b2be1250c7f4a0a4c98f20a2b1

**Pull request merged**